src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -143,17 +146,19 @@ public override void WriteCData(string? text)
             WriteString(text);
         }
 
-        // Forward call to WriteString(string).
+        // Forward call to WriteChars.
         public override void WriteCharEntity(char ch)
         {
-            WriteString(char.ToString(ch));
+            _primitivesBuffer[0] = ch;
```

Comment:
Looks like this is not correct.
char.ToString(ch) is not the same as just assigning char to buffer.
Am I right ?

Also there is method in XmlConverter  that is doing same 
```
public static string ToString(char value)
  {
      return value.ToString();
  }
```

but serializers are using something different :
```
protected static string FromChar(char value)
  {
      return XmlCustomFormatter.FromChar(value);
  }
```
XmlCustomFormatter:
```
internal static string FromChar(char value)
  {
      return XmlConvert.ToString((ushort)value);
  }
```

Maybe it is a good idea somehow to unify this....

Reply:
> char.ToString(ch) is not the same as just assigning char to buffer.

Looking at the implementation the two appear to be equivalent:

https://github.com/dotnet/runtime/issues/75327#issuecomment-1246401495

It might make sense to compare with the other built-in Xml implementations before making the change.

Reply:
I choose to mimic XmlCustomFormatter code with XmlConvert.ToString((ushort)value);